notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/NotificationHubInstallationAdapter.java

```diff
@@ -69,6 +73,16 @@ public void onResponse(String response) {
                     @Override
                     public void onErrorResponse(VolleyError error) {
                         Log.e("ANH", "Couldn't update installation: " + error);
+                        String retryAfter = error.networkResponse.headers.get("Retry-After");

```

Comment:
Instead of this approach, could we go with an exponential backoff that retries a number of times and then gives up?  Look at the retrier from AppCenter for inspiration: https://github.com/microsoft/appcenter-sdk-android/blob/develop/sdk/appcenter/src/main/java/com/microsoft/appcenter/http/HttpClientRetryer.java

Reply:
We need both retry strategies, each for a different situation. General transient errors like lack of connectivity should be retried with an exponential back-off. In situations like below where we see a Retry-After header (specifically on a 503 or 429 response), we should respect the server's suggestion and wait that amount of time.

I think your broader point is that we probably don't need to reinvent the wheel here, because App Center has logic we could use. Similarly, we could pull in an AutoRest generated library which would accomplish the same thing. To me, the most important thing is that we don't leak our solution, whatever we choose, and end up having more public surface area to support without having a specific design review. That was always the appeal of having an inline bespoke client. But there are other approaches.

Reply:
The AppCenter SDK does cover both scenarios for the 429/503, but also the retry-after header.  But we need to recover from 403 errors, which is what our current V1 API returns when throttled.

Reply:
Yep, thanks
I've ported these http package to our SDK since we can't just get it from Maven - some custom logic needs to be applied (like there is retry only for POST from the box)